File: MiniDOMOffice/CRM/CCampagnaCRM/CCampagnaCRMCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases

Imports minidom.Anagrafica

Partial Public Class CustomerCalls


    Public Class CCampagnaCRMCursor
        Inherits DBObjectCursorPO(Of CCampagnaCRM)

        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_Inizio As New CCursorField(Of Date)("Inizio")
        Private m_Fine As New CCursorField(Of Date)("Fine")
        Private m_TipoAssegnazione As New CCursorField(Of TipoCampagnaCRM)("TipoAssegnazione")
        Private m_TipoCampagna As New CCursorFieldObj(Of String)("TipoCampagna")
        Private m_Flags As New CCursorField(Of CampagnaCRMFlag)("Flags")

        Public Sub New()

        End Sub

        Public ReadOnly Property Inizio As CCursorField(Of Date)
            Get
                Return Me.m_Inizio
            End Get
        End Property

        Public ReadOnly Property Fine As CCursorField(Of Date)
            Get
                Return Me.m_Fine
            End Get
        End Property

        Public ReadOnly Property TipoAssegnazione As CCursorField(Of TipoCampagnaCRM)
            Get
                Return Me.m_TipoAssegnazione
            End Get
        End Property

        Public ReadOnly Property TipoCampagna As CCursorFieldObj(Of String)
            Get
                Return Me.m_TipoCampagna
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of CampagnaCRMFlag)
            Get
                Return Me.m_Flags
            End Get
        End Property


        Protected Overrides Function GetModule() As CModule
            Return CustomerCalls.CampagneCRM.Module
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return CustomerCalls.CRM.StatsDB
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_CampagnaCRM"
        End Function


    End Class



End Class

File: MiniDOMOffice/CRM/CCampagnaCRM/CCampagnaCRMCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class CustomerCalls
    {
        /// <summary>
        /// Cursore sulla tabella <see cref="CCampagnaCRM"/>
        /// </summary>
        [Serializable]
        public class CCampagnaCRMCursor 
            : Databases.DBObjectCursorPO<CCampagnaCRM>
        {
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorField<DateTime> m_Inizio = new DBCursorField<DateTime>("Inizio");
            private DBCursorField<DateTime> m_Fine = new DBCursorField<DateTime>("Fine");
            private DBCursorField<TipoCampagnaCRM> m_TipoAssegnazione = new DBCursorField<TipoCampagnaCRM>("TipoAssegnazione");
            private DBCursorStringField m_TipoCampagna = new DBCursorStringField("TipoCampagna");
            private DBCursorField<CampagnaCRMFlag> m_Flags = new DBCursorField<CampagnaCRMFlag>("Flags");

            /// <summary>
            /// Costruttore
            /// </summary>
            public CCampagnaCRMCursor()
            {
            }

            /// <summary>
            /// Inizio
            /// </summary>
            public DBCursorField<DateTime> Inizio
            {
                get
                {
                    return m_Inizio;
                }
            }

            /// <summary>
            /// Fine
            /// </summary>
            public DBCursorField<DateTime> Fine
            {
                get
                {
                    return m_Fine;
                }
            }

            /// <summary>
            /// TipoAssegnazione
            /// </summary>
            public DBCursorField<TipoCampagnaCRM> TipoAssegnazione
            {
                get
                {
                    return m_TipoAssegnazione;
                }
            }

            /// <summary>
            /// TipoCampagna
            /// </summary>
            public DBCursorStringField TipoCampagna
            {
                get
                {
                    return m_TipoCampagna;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<CampagnaCRMFlag> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.CustomerCalls.CampagneCRM;
            }
             
        }
    }
}